contrib/lua/premake5.lua

```diff
@@ -21,8 +21,8 @@ project "lua-lib"
 		"etc/*.c"
 	}
 
-	configuration "linux or bsd or hurd or aix"
+	filter "system:linux or system:bsd or system:hurd or system:aix"
```

Comment:
I find the fact that it only applies to the first term to be a bug.

I honestly don't know if adding support for parenthesis is a good idea, it increases the complexity of the code that needs to be maintained. Also, it's just waiting for someone to request `and` support, which will make having a list of strings with an implicit `and` kind of redundant. I feel like we should just use a function at that point, but that's probably just going to make it harder to use filter.